File: article-admin-main/src/types/transfer.ts
export interface TransferTargetConfig {
  host: string
  port: number
  database: string
  username: string
  password: string
  schema: string
  table: string
  schedule_enabled: boolean
  schedule_cron: string
}

export interface TransferTablesResult {
  schema: string
  tables: string[]
}

export interface TransferRunResult {
  schema: string
  table: string
  total: number
  inserted: number
  skipped: number
}

File: article-master/app/schemas/transfer.py
from pydantic import BaseModel, Field


class TransferTargetPayload(BaseModel):
    host: str = ""
    port: int = Field(default=5432, ge=1)
    database: str = ""
    username: str = ""
    password: str = ""
    schema: str = "public"
    table: str = ""
    schedule_enabled: bool = False
    schedule_cron: str = "0 2 * * *"


class TransferTableLookupPayload(BaseModel):
    host: str = ""
    port: int = Field(default=5432, ge=1)
    database: str = ""
    username: str = ""
    password: str = ""
    schema: str = "public"
